pkg/chains/annotations/annotations.go

```diff
@@ -148,3 +154,53 @@ func AddAnnotation(ctx context.Context, obj objects.TektonObject, ps versioned.I
 
 	return nil
 }
+
+// HandleRetry handles retries managed as annotation on the k8s object
+func HandleRetry(ctx context.Context, obj objects.TektonObject, ps versioned.Interface, annotationsMap map[string]string) error {
+	if RetryAvailable(obj) {
+		return AddRetry(ctx, obj, ps, annotationsMap)
+	}
+	return MarkFailed(ctx, obj, ps, annotationsMap)
+}
+
+// CreateAnnotationsPatch returns patch bytes that can be used with kubectl patch
+func CreateAnnotationsPatch(newAnnotations map[string]string, obj objects.TektonObject) ([]byte, error) {
+	// Get GVK using the TektonObject interface method (more reliable than runtime.Object)
+	gvkStr := obj.GetGVK()
+	if gvkStr == "" {
+		return nil, fmt.Errorf("unable to determine GroupVersionKind for object %s/%s", obj.GetNamespace(), obj.GetName())
+	}
+
+	// Parse the string format "group/version/kind"
+	parts := strings.Split(gvkStr, "/")
+	if len(parts) != 3 {
```

Comment:
```suggestion
	if len(parts) != 3 || parts[0] == "" || parts[1] == "" || parts[2] == "" {
```

just to make sure that we don't have silent failures because of the string concatenation done on the subsequent lines

Reply:
added